src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java

```diff
@@ -230,6 +231,15 @@ public synchronized void run() {
          */
         @Override
         public void run() {
+            while (!ProtocolStack.isStarted().get()) {
```

Comment:
You could always use a CountDownLatch that counts down when transitioning to started. Then this method can call await rather than busy-waiting with sleep.

Reply:
Good idea. That's an improved implementation.